akka-http-core/src/main/java/akka/http/javadsl/model/HttpHeader.java

```diff
@@ -4,11 +4,14 @@
 
 package akka.http.javadsl.model;
 
+import akka.annotation.DoNotInherit;
+
 /**
  * The base type representing Http headers. All actual header values will be instances
  * of one of the subtypes defined in the `headers` packages. Unknown headers will be subtypes
  * of {@link akka.http.javadsl.model.headers.RawHeader}.
  */
+@DoNotInherit
```

Comment:
figured might as well mark them as such since i was looking around for some things